common/scala/src/main/scala/whisk/core/entity/WhiskAction.scala

```diff
@@ -330,18 +330,24 @@ object WhiskAction extends DocumentFactory[WhiskAction] with WhiskEntityQueries[
           implicit val logger = db.logging
           implicit val ec = db.executionContext
 
-          val newDoc = doc.copy(exec = exec.attach)
-          newDoc.revision(doc.rev)
-
           val stream = new ByteArrayInputStream(Base64.getDecoder().decode(code))
           val manifest = exec.manifest.attached.get
-
-          for (i1 <- super.put(db, newDoc);
-               i2 <- attach[A](db, newDoc.revision(i1.rev), manifest.attachmentName, manifest.attachmentType, stream))
-            yield i2
```

Comment:
Not sure but probably existing flow had a race condition where the `super.put(db, newDoc)` would have "published" a cache value which is yet not complete. If any thread see this value and tries to read the attachment the call would fail. So may be cache should only be updated with attach